homeassistant/components/kostal_plenticore/select.py

```diff
@@ -100,36 +97,33 @@ class PlenticoreDataSelect(CoordinatorEntity, SelectEntity, ABC):
     """Representation of a Plenticore Select."""
 
     _attr_entity_category = EntityCategory.CONFIG
+    entity_description: PlenticoreSelectEntityDescription
 
     def __init__(
         self,
-        coordinator,
+        coordinator: SelectDataUpdateCoordinator,
+        description: PlenticoreSelectEntityDescription,
         entry_id: str,
         platform_name: str,
         device_class: str | None,
-        module_id: str,
-        data_id: str,
-        name: str | None,
         current_option: str | None,
-        options: list[str],
-        is_on: str,
         device_info: DeviceInfo,
-        unique_id: str,
     ) -> None:
         """Create a new Select Entity for Plenticore process data."""
         super().__init__(coordinator)
+        self.entity_description = description
         self.entry_id = entry_id
         self.platform_name = platform_name
         self._attr_device_class = device_class
-        self.module_id = module_id
-        self.data_id = data_id
-        self._attr_options = options
-        self.all_options = options
+        self.module_id = description.module_id
+        self.data_id = description.key
+        self._attr_options = description.options
```

Comment:
Side note: Would be nice to extend the SelectEntityDescription with an attribute `options`.

Reply:
I thought about it, but I then I thought maybe it was left out on purpose.
I'll take a fresh look later...